packages/utils/src/dal/mikro-orm/integration-tests/__tests__/mikro-orm-repository.spec.ts

```diff
@@ -0,0 +1,844 @@
+        id: "1",
+        title: "en1",
+        entity3: [{ title: "en3-1" }, { title: "en3-2" }],
+      }
+
+      await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: ["entity3"],
+        relationsToSkip: [],
+      })
+      const listedEntity1 = await manager1().find({
+        where: { id: "1" },
+        options: { populate: ["entity3"] },
+      })
+
+      const listedEntity3 = await manager3().find({
+        where: { title: "en3-1" },
+        options: { populate: ["entity1"] },
+      })
+
+      expect(listedEntity1).toHaveLength(1)
+      expect(listedEntity1[0]).toEqual(
+        expect.objectContaining({
+          id: "1",
+          title: "en1",
+        })
+      )
+      expect(listedEntity1[0].entity3.getItems()).toHaveLength(2)
+      expect(listedEntity1[0].entity3.getItems()).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            title: "en3-1",
+          }),
+          expect.objectContaining({
+            title: "en3-2",
+          }),
+        ])
+      )
+
+      expect(listedEntity3).toHaveLength(1)
+      expect(listedEntity3[0]).toEqual(
+        expect.objectContaining({
+          title: "en3-1",
+        })
+      )
+      expect(listedEntity3[0].entity1.getItems()).toHaveLength(1)
+      expect(listedEntity3[0].entity1.getItems()).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            title: "en1",
+          }),
+        ])
+      )
+    })
+
+    it("should only update the parent entity of a many-to-many if relation is not included", async () => {
+      const entity1 = {
+        id: "1",
+        title: "en1",
+        entity3: [{ title: "en3-1" }, { title: "en3-2" }],
+      }
+
+      await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: ["entity3"],
+        relationsToSkip: [],
+      })
+      entity1.title = "newen1"
+      entity1.entity3.push({ title: "en3-3" })
+      await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: [],
+        relationsToSkip: ["entity3"],
+      })
+
+      const listedEntities = await manager1().find({
+        where: { id: "1" },
+        options: { populate: ["entity3"] },
+      })
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0].title).toEqual("newen1")
+
+      expect(listedEntities[0].entity3.getItems()).toHaveLength(2)
+      expect(listedEntities[0].entity3.getItems()).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            title: "en3-1",
+          }),
+          expect.objectContaining({
+            title: "en3-2",
+          }),
+        ])
+      )
+    })
+
+    it("should successfully update, create, and delete subentities an entity with a many-to-many relation", async () => {
+      const entity1 = {
+        id: "1",
+        title: "en1",
+        entity3: [
+          { id: "4", title: "en3-1" },
+          { id: "5", title: "en3-2" },
```

Comment:
q: should updating actually be allowed in a many to many? 

The entity3 might be a part of several entities, updating through this entity will also affect all the other entities that its associated with.

Reply:
that is true, we now only allow to create, attach an existing, delete

Reply:
That's a good point, let me think about it.

Reply:
@riqwan added a comment but I won't address this for now, as ultimately the control is in the caller if they want to pass additional data or not.